bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -229,7 +229,13 @@ func (ms *ManagedStream) openWithRetry() (storagepb.BigQueryWrite_AppendRowsClie
 		if err == nil {
 			// The channel relationship with its ARC is 1:1.  If we get a new ARC, create a new chan
 			// and fire up the associated receive processor.
-			ch := make(chan *pendingWrite)
+			depth := 1000 // default backend queue limit
```

Comment:
Does each call to receive guaranteed to only move the offset by one? Lets say the offset is N. We send two request to append rows before we receive. Will receive always be N+1 then N+2? Or could it bump to N+2 right away? Just want to make sure we will not build up too many requests that can never be processed with this new paradigm.

Reply:
its based on the rows, not the number of calls.  append can send N rows per call.

Example:
*  ​append 4 rows @ offset 0: offset is now 4 (0,1,2,3 written)
* append 2 rows @ offset 0: error, already exists
* append 2 rows @ offset 4: offset is now 6